packages/dds/map/src/directory.ts

```diff
@@ -943,10 +978,16 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
     private readonly pendingKeys: Map<string, number[]> = new Map();
 
     /**
-     * Subdirectories that have been modified locally but not yet ack'd from the server.
+     * Subdirectories that have been created/deleted locally but not yet ack'd from the server.
      */
     private readonly pendingSubDirectories: Map<string, number[]> = new Map();
 
+    /**
+     * Subdirectories that have been deleted locally but not yet ack'd from the server. This maintains the count
+     * of delete op that are pending or yet to be acked from server.
+     */
+    private readonly pendingDeleteSubDirectoriesCount: Map<string, number> = new Map();
```

Comment:
I think it should be fine, but can you please ensure all of this logic and new state tracking works Ok when we serialize container with pending deletes and later on rehydrate it? I think we need to have UTs that validate things work. I guess there is no special handling required, as we replay changes (i.e. essentially generate appropriate changes again), but better be tested :)

Reply:
In detached container or when we serialize it then there are no ops in that state. So, there will be no pending deletes, right?